### ISSUE TITLE
In the member function ParseNextKey() of the Block::Ite class in the leveldb/table/block.cc file, the last member variable restart_index_ updated.

### ISSUE THREAD

**Issue by LunarStore** [google/leveldb#1138]
In the leveldb/table/block.cc file, the member function ParseNextKey() of the Block::Ite class, when updating the member variable restart_index_, the last judgment condition of the while loop (**GetRestartPoint(restart_index_ + 1) < current_**) should be added with an equal sign, right? If the first KV of the restart point is read without the equal sign, the restart_index will not be updated in time, and the second KV must be read to update.

**Comment by zjs1224522500** [comment]
Yes, you are right! The restart_index is not be updated in time for the first entry, but I found it had little effect. 

I'm not sure if introduce any new bugs when we make changes to this condition `(GetRestartPoint(restart_index_ + 1) < current_)`. Perhaps you can run some tests to verify that it is correct and create a PR.
